var/spack/repos/builtin/packages/singularity/package.py

```diff
@@ -30,8 +30,9 @@ class Singularity(AutotoolsPackage):
        Compute'"""
 
     homepage = "http://singularity.lbl.gov/"
-    url      = "https://github.com/singularityware/singularity/archive/2.3.1.tar.gz"
+    url      = "https://github.com/singularityware/singularity/archive/2.4.tar.gz"
 
+    version('2.4', 'd357ce68ef2f8149edd84155731531465dbe74148c37719f87f168fc39384377')
```

Comment:
Why is that checksum so long?

Reply:
that's a SHA256.  We'll be switching everything to these anyway, so have at it I say.